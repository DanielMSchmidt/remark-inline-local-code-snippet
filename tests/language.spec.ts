import { describe, it, expect } from "vitest";
import {
  supportedLanguageExtensions,
  pathExtensionToMarkdownLanguageTag,
  commentOutBasedOnLanguage,
} from "../src/index";

describe("Languages", () => {
  describe.each(Object.keys(supportedLanguageExtensions))("supported - %s", (ext) => {
    it("has a markdown mapping", () => {
      expect(pathExtensionToMarkdownLanguageTag(ext)).toMatchSnapshot();
    });
    it("can comment out code", () => {
      expect(commentOutBasedOnLanguage(ext, "code")).toMatchSnapshot();
    });
  });
  describe("unsupported", () => {
    const ext = ".unsupported";
    it("has a markdown mapping", () => {
      expect(pathExtensionToMarkdownLanguageTag(ext)).toMatchSnapshot();
    });
    it("can comment out code", () => {
      expect(commentOutBasedOnLanguage(ext, "code")).toMatchSnapshot();
    });
  });
});
