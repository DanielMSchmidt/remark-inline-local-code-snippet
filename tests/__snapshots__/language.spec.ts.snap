// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Languages > supported - .go > can comment out code 1`] = `
"// code
"
`;

exports[`Languages > supported - .go > has a markdown mapping 1`] = `"go"`;

exports[`Languages > supported - .hcl > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .hcl > has a markdown mapping 1`] = `"hcl"`;

exports[`Languages > supported - .js > can comment out code 1`] = `
"// code
"
`;

exports[`Languages > supported - .js > has a markdown mapping 1`] = `"javascript"`;

exports[`Languages > supported - .json > can comment out code 1`] = `""`;

exports[`Languages > supported - .json > has a markdown mapping 1`] = `"json"`;

exports[`Languages > supported - .py > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .py > has a markdown mapping 1`] = `"python"`;

exports[`Languages > supported - .sh > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .sh > has a markdown mapping 1`] = `"bash"`;

exports[`Languages > supported - .tf > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .tf > has a markdown mapping 1`] = `"terraform"`;

exports[`Languages > supported - .tfstacks.hcl > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .tfstacks.hcl > has a markdown mapping 1`] = `"hcl"`;

exports[`Languages > supported - .ts > can comment out code 1`] = `
"// code
"
`;

exports[`Languages > supported - .ts > has a markdown mapping 1`] = `"typescript"`;

exports[`Languages > supported - .yaml > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .yaml > has a markdown mapping 1`] = `"yaml"`;

exports[`Languages > supported - .yml > can comment out code 1`] = `
"# code
"
`;

exports[`Languages > supported - .yml > has a markdown mapping 1`] = `"yaml"`;

exports[`Languages > unsupported > can comment out code 1`] = `
"// code
"
`;

exports[`Languages > unsupported > has a markdown mapping 1`] = `""`;
